xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/AttachmentWikiResource.java

```diff
@@ -56,4 +59,20 @@ public String getURL(XWikiDocument document) throws Exception
     {
         return document.getAttachmentURL(this.reference.getName(), "skin", this.xcontextProvider.get());
     }
+
+    @Override
+    public String getURL(XWikiDocument document, Map<String, String> queryParameters) throws Exception
+    {
+        StringBuilder queryString = new StringBuilder();
+
+        for (Map.Entry<String, String> parameters : queryParameters.entrySet()) {
```

Comment:
I'm sure this query string generation can go (or even already exist) in some common utility class.